### PR TITLE
global: addition of tests for REST API

### PR DESCRIPTION
**Pull request by PaulinaLach**
- NEW Adds tests for module's REST API. ( #1 )

Signed-off-by: PaulinaLach paulina.malgorzata.lach@cern.ch

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -39,3 +40,53 @@ def app():
...
+            password='pass1',
+            active=False
+        ),
+    ]
+
+
+@pytest.fixture()
+def users(db, users_data):
+    """Create test users."""
+    return [
```
Comment (jirikuncar) [inline tests/conftest.py]: `return [create_test_user(**user) for user in users_data]`

File: tests/conftest.py [hunk 2]
```diff
@@ -39,3 +40,53 @@ def app():
...
+    ]
+
+
+@pytest.fixture()
+def roles(db, roles_data):
+    """Create test roles."""
+    ds = app.extensions['invenio-accounts'].datastore
+
+    with app.app_context():
+        r1 = ds.create_role(**roles_data[0])
```
Comment (jirikuncar) [inline tests/conftest.py]:
``` python
result = [ds.create_role(**role) for role in roles_data]
ds.commit()

# ...
return result
```

File: .travis.yml [hunk 3]
```diff
@@ -40,6 +40,8 @@ env:
 
 python:
   - "2.7"
+  - "3.3"
+  - "3.4"
```
Comment (tiborsimko) [inline .travis.yml]: FWIW we decided some time ago to support Python 2.7 and 3.5 only, see https://github.com/inveniosoftware/invenio/issues/3647. Is Python 3.3 or 3.4 needed for B2SHARE? CC @nharraud

File: invenio_accounts_rest/views.py [hunk 4]
```diff
@@ -81,3 +237,67 @@ def get(self):
...
+blueprint.add_url_rule(
+    '/roles/<string:role_id>',
+    view_func=RoleResource.as_view(
+        RoleResource.view_name
+    )
+)
+
+
+blueprint.add_url_rule(
+    '/roles/<string:role_id>/users/<string:user_id>/assign',
```
Comment (nharraud) [inline invenio_accounts_rest/views.py]: PUT '/roles/string:role_id/users/string:user_id'
would be more REST (no verb in the URL).

File: invenio_accounts_rest/views.py [hunk 5]
```diff
@@ -81,3 +237,67 @@ def get(self):
...
+    view_func=AssignRoleResource.as_view(
+        AssignRoleResource.view_name
+    )
+)
+
+
+blueprint.add_url_rule(
+    '/roles/<string:role_id>/users/<string:user_id>/unassign',
```
Comment (nharraud) [inline invenio_accounts_rest/views.py]: DELETE '/roles/string:role_id/users/string:user_id'
would be more REST (no verb in the URL).

File: invenio_accounts_rest/views.py [hunk 6]
```diff
@@ -81,3 +237,67 @@ def get(self):
...
+blueprint.add_url_rule(
+    '/users/<string:user_id>/roles',
+    view_func=UserRolesListResource.as_view(
+        UserRolesListResource.view_name
+    )
+)
+
+
+blueprint.add_url_rule(
+    '/users/<string:user_id>/properties',
```
Comment (nharraud) [inline invenio_accounts_rest/views.py]: GET '/users/string:user_id'
should be enough IMHO. Or is it conflicting with some other endpoint.

**Review by nharraud — CHANGES_REQUESTED** [review]
Only endpoint changes. Otherwise it looks good for a test-only PR.

**Comment by nharraud** [comment]
closing in favor of #7

**Pull request closed without merging**